managed/services/management/agent_test.go

```diff
@@ -0,0 +1,183 @@
+
+	ctx := logger.Set(context.Background(), t.Name())
+	uuid.SetRand(&tests.IDReader{})
+
+	sqlDB := testdb.Open(t, models.SetupFixtures, nil)
+	db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+
+	vmdb := &mockPrometheusService{}
+	vmdb.Test(t)
+
+	state := &mockAgentsStateUpdater{}
+	state.Test(t)
+
+	ar := &mockAgentsRegistry{}
+	ar.Test(t)
+
+	teardown := func(t *testing.T) {
+		uuid.SetRand(nil)
+
+		require.NoError(t, sqlDB.Close())
+		vmdb.AssertExpectations(t)
+		state.AssertExpectations(t)
+		ar.AssertExpectations(t)
+	}
+	s := NewAgentService(db, ar)
+
+	return ctx, s, teardown, vmdb
+}
+
+func TestAgentService(t *testing.T) {
+	t.Run("List", func(t *testing.T) {
+		t.Run("Basic", func(t *testing.T) {
+			ctx, s, teardown, _ := setup(t)
+			defer teardown(t)
+
+			services, err := models.FindServices(s.db.Querier, models.ServiceFilters{})
+			require.NoError(t, err)
+			assert.Len(t, services, 1)
+			service := services[0]
+
+			s.r.(*mockAgentsRegistry).On("IsConnected", models.PMMServerAgentID).Return(true).Once()                           // PMM Server Agent
+			s.r.(*mockAgentsRegistry).On("IsConnected", "/agent_id/00000000-0000-4000-8000-000000000003").Return(false).Once() // PMM Server PostgreSQL exporter
+			s.r.(*mockAgentsRegistry).On("IsConnected", "/agent_id/00000000-0000-4000-8000-000000000004").Return(false).Once() // PMM Server PG Stat Statements agent
+			response, err := s.ListAgents(ctx, &managementpb.ListAgentRequest{
+				ServiceId: service.ServiceID,
+			})
+
+			require.NoError(t, err)
+			assert.Len(t, response.Agents, 3) // 2 exporters + 1 agent
+		})
+
+		t.Run("RDS", func(t *testing.T) {
+			ctx, s, teardown, _ := setup(t)
+			defer teardown(t)
+
+			node, err := models.CreateNode(s.db.Querier, models.RemoteRDSNodeType, &models.CreateNodeParams{
+				NodeName: "test",
+				Address:  "test-address",
+				Region:   pointer.ToString("test-region"),
+			})
+			require.NoError(t, err)
+
+			service, err := models.AddNewService(s.db.Querier, models.MySQLServiceType, &models.AddDBMSServiceParams{
+				ServiceName: "test-mysql",
+				NodeID:      node.NodeID,
+				Address:     pointer.ToString("127.0.0.1"),
+				Port:        pointer.ToUint16(3306),
+			})
+			require.NoError(t, err)
+
+			pmmAgent, err := models.CreatePMMAgent(s.db.Querier, models.PMMServerNodeID, nil)
+			require.NoError(t, err)
+
+			mysqldExporter, err := models.CreateAgent(s.db.Querier, models.MySQLdExporterType, &models.CreateAgentParams{
+				PMMAgentID: pmmAgent.AgentID,
+				ServiceID:  service.ServiceID,
+				Password:   "password",
+				Username:   "username",
+			})
+			require.NoError(t, err)
+
+			rdsExporter, err := models.CreateAgent(s.db.Querier, models.RDSExporterType, &models.CreateAgentParams{
+				PMMAgentID: pmmAgent.AgentID,
+				ServiceID:  service.ServiceID,
+			})
+			require.NoError(t, err)
+
+			s.r.(*mockAgentsRegistry).On("IsConnected", mysqldExporter.AgentID).Return(false) // MySQLd exporter
```

Comment:
```suggestion
			s.r.(*mockAgentsRegistry).On("IsConnected", mysqldExporter.AgentID).Return(false).Once() // MySQLd exporter
```

And add the same for all the assumed calls in this file and `service_test.go`

Reply:
Done.